torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -267,6 +267,39 @@ def call_function(
         unimplemented(f"HigherOrderOperator {self.value.__name__}")
 
 
+# We rely on the "example_value" of lifted_proxies to
+# get the source of the corresponding TensorVariable so
+# as to associate it with the python-level variables.
+# TODO: maybe it's better to get a reverse mapping between
+# proxy to VariableTracker.
```

Comment:
To summarize my understanding of the situation: the claim is that the free variable returns are a bit sketchy. Our free variable system notices if a Proxy is a free variable. Somehow we need to relate this back to a Tensor, and the way we're doing it is using proxy.meta['fake'], looking up tx.tracked_fakes, finding the name of the Tensor, and then eval'ing it.

Need to think more about if there's something less sketchy we can do here